src/components/TodosFilter/TodosFilter.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useMemo } from 'react';
+import classNames from 'classnames';
+import { FilterStatus } from '../../types/FilterStatus';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[],
+  setFilterType: (value: FilterStatus) => void,
+  filterType: FilterStatus,
+  onRemove: () => void;
+};
+
+export const TodosFilter: React.FC<Props> = ({
+  todos,
+  setFilterType,
+  filterType,
+  onRemove,
+}) => {
+  const todosLeft = useMemo(() => {
+    return todos.filter(todo => !todo.completed);
+  }, []).length;
+
+  const todosCompleted = todos.length - todosLeft;
+
+  if (todos.length < 1) {
+    return null;
+  }
+
+  return (
+    <footer
+      className="todoapp__footer"
+      data-cy="Footer"
+    >
+      <span className="todo-count" data-cy="todosCounter">
+        {`${todosLeft} items left`}
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          data-cy="FilterLinkAll"
+          href="#/"
+          onClick={() => setFilterType(FilterStatus.All)}
+          className={classNames(
+            'filter__link',
+            {
+              selected: filterType === FilterStatus.All,
+            },
+          )}
+        >
+          All
+        </a>
+
+        <a
+          data-cy="FilterLinkActive"
+          href="#/active"
+          onClick={() => setFilterType(FilterStatus.Active)}
+          className={classNames(
+            'filter__link',
+            {
+              selected: filterType === FilterStatus.Active,
+            },
+          )}
+        >
+          Active
+        </a>
+        <a
+          data-cy="FilterLinkCompleted"
+          href="#/completed"
+          onClick={() => setFilterType(FilterStatus.Completed)}
+          className={classNames(
+            'filter__link',
+            {
+              selected: filterType === FilterStatus.Completed,
+            },
+          )}
+        >
+          Completed
+        </a>
+      </nav>
+
+      <button
+        data-cy="ClearCompletedButton"
+        onClick={onRemove}
+        type="button"
+        className="todoapp__clear-completed"
+      >
+        {(todosCompleted > 0) && 'Clear completed'}
```

Comment:
Check how this button works on [the working example](https://mate-academy.github.io/react_todo-app-with-api/) and yours.
<img width="904" alt="Screenshot 2022-10-10 at 19 59 08" src="https://user-images.githubusercontent.com/17098736/194918273-358997cc-2859-4833-a3a8-6eb779931958.png">